src/libmongoc/src/mongoc/mongoc-cursor.c

```diff
@@ -1728,9 +1729,13 @@ _mongoc_cursor_response_refresh (mongoc_cursor_t *cursor,
    /* server replies to find / aggregate with {cursor: {id: N, firstBatch: []}},
     * to getMore command with {cursor: {id: N, nextBatch: []}}. */
    if (_mongoc_cursor_run_command (
-          cursor, command, opts, &response->reply, false) &&
-       _mongoc_cursor_start_reading_response (cursor, response)) {
-      return;
+          cursor, command, opts, &response->reply, false)) {
+      if (_mongoc_cursor_start_reading_response (cursor, response)) {
+         // TODO here or in above function: read moreToCome bit from server
+         // set cursor->in_exhaust and cursor->client->in_exhaust accordingly
```

Comment:
Suggest creating a JIRA ticket for this TODO if one does not already exist + referencing the ticket number in this comment.